File: src/main/java/com/nicholasnassar/sfsapi/data/gpa/GPACalculation.java
package com.nicholasnassar.sfsapi.data.gpa;

import io.vertx.core.json.JsonArray;
import io.vertx.core.json.JsonObject;

import java.util.List;

public class GPACalculation {
    private final List<GPAClass> classes;

    private final String[] selectableTerms;

    public GPACalculation(List<GPAClass> classes, String[] selectableTerms) {
        this.classes = classes;

        this.selectableTerms = selectableTerms;
    }

    public JsonObject asJson() {
        JsonArray array = new JsonArray();

        for (GPAClass clazz : classes) {
            array.add(clazz.asJson());
        }

        JsonArray selectableTermsArray = new JsonArray();

        for (String previousTerm : selectableTerms) {
            selectableTermsArray.add(previousTerm);
        }

        return new JsonObject().put("classes", array).put("selectable_terms", selectableTermsArray);
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/gpa/GPACalculation.kt
package com.nicholasnassar.sfsapi.data.gpa

import io.vertx.core.json.JsonArray
import io.vertx.core.json.JsonObject

class GPACalculation(val classes: List<GPAClass>, val selectableTerms: Array<String>) {
    fun asJson(): JsonObject {
        val array = JsonArray()

        for (clazz in classes) {
            array.add(clazz.asJson())
        }

        val selectableTermsArray = JsonArray()

        for (previousTerm in selectableTerms) {
            selectableTermsArray.add(previousTerm)
        }

        return JsonObject().put("classes", array).put("selectable_terms", selectableTermsArray)
    }
}